### PR TITLE
ui: change is-recommended icon to svg

### PR DESCRIPTION
**Pull request by seanfarley**
This is just a minor clean up to get a fix out there before this code bit rots in my fork :-)

### CODE REVIEW

File: exercism-modern.el [hunk 1]
```diff
@@ -245,13 +245,12 @@ Pass prefix BUFFER-PREFIX-ARG to prompt for a buffer instead."
                                            (blurb (alist-get 'blurb exercise))
                                            (difficulty (alist-get 'difficulty exercise))
                                            (is-unlocked (not (eq :json-false (alist-get 'is_unlocked exercise))))
-                                           (is-recommended (not (eq :json-false (alist-get 'is_recommended exercise))))
```
Comment (elken) [inline exercism-modern.el:248]: Why removed?

File: exercism-modern.el [hunk 2]
```diff
@@ -245,13 +245,12 @@ Pass prefix BUFFER-PREFIX-ARG to prompt for a buffer instead."
...
                                            (text-face (if is-unlocked 'default 'shadow))
-                                           (foreground (face-attribute (intern (format "exercism-%s-button" difficulty)) :foreground))
-                                           (background (face-attribute (intern (format "exercism-%s-button" difficulty)) :background)))
+                                           (foreground (face-attribute (intern (format "exercism-modern-%s-button" difficulty)) :foreground))
+                                           (background (face-attribute (intern (format "exercism-modern-%s-button" difficulty)) :background)))
                                       (list slug
                                             (vector (concat
-                                                     (propertize "  " 'face 'warning)
+                                                     (propertize "★  " 'face 'warning)
```
Comment (elken) [inline exercism-modern.el:253]: How does this look with font scaling? Massive/small fonts?

File: exercism-modern.el (hunk 1 above)
Comment (seanfarley) [reply exercism-modern.el:248]: Because the linter said it was unused? (and because it was not used anywhere else)

File: exercism-modern.el (hunk 2 above)
Comment (seanfarley) [reply exercism-modern.el:253]: They scale with the font:

Screen Shot 2022-10-06 at 7 46 38 PM

The svg wasn't so lucky

**Comment by seanfarley** [comment]
> I'd also prefer if these better matched doom's conventional commits, I doubt I'll enforce CI for them unless this gets large usage but I'd rather they match those :D

Might be fairly easy to do that after doom cli 3.0, right? In fact, once `doom ci` is out, I'll probably switch to it as well in my own projects.

File: exercism-modern.el (hunk 2 above)
Comment (elken) [reply exercism-modern.el:253]: That's good, annoying about the svgs though.

Can you raise an issue for that?

File: exercism-modern.el (hunk 1 above)
Comment (elken) [reply exercism-modern.el:248]: Oh that's a bug, the star should only be shown for recommended exercises

**Comment by elken** [comment]
Maybe, but for now I'm not _too_ fussed.

Just rebase them to better match :D

**Comment by seanfarley** [comment]
Sure, what kind of wording would you prefer?

**Comment by elken** [comment]
> Sure, what kind of wording would you prefer?

The doom style of conventional commits :P

**Comment by seanfarley** [comment]
> > Sure, what kind of wording would you prefer?
> 
> The doom style of conventional commits :P

Right but doom has module's and distinct separation with different areas of code. What would be a list of "modules" or "areas" for this?

**Comment by elken** [comment]
It's pretty simple so I guess function names are fine for scopes

**Comment by seanfarley** [comment]
Ok, I'll try that

File: exercism-modern.el (hunk 1 above)
Comment (elken) [reply exercism-modern.el:248]: Resolved in 4cdb41c208dda18386f383fbf0333ea7155463d2

**Comment by elken** [comment]
@seanfarley you'll need to rebase to pick up some new changes (sorry!)

**Comment by elken** [comment]
I wonder if that should also be an svg now

**Comment by seanfarley** [comment]
After the rebase, the recommended star is a bit off. Should there be an empty space added when it's not recommended?
Screen Shot 2022-10-10 at 8 57 51 AM

**Comment by seanfarley** [comment]
I just whipped this up but the names are not lining up with the header "Exercise"

Screen Shot 2022-10-10 at 9 01 32 AM

**Comment by seanfarley** [comment]
> > I just whipped this up but the names are not lining up with the header "Exercise"
> > Screen Shot 2022-10-10 at 9 01 32 AM
> 
> The header would also need a similar amount of padding, unless we add a column with no heading to put the star in otherwise blank

Yeah, I was just whipping that up now to test what it looked like

**Comment by seanfarley** [comment]
The header being a star or blank is up to you:

Screen Shot 2022-10-10 at 9 21 24 AM

Follow-up question: should this star just be a svg like the other icons?

**Comment by elken** [comment]
> The header being a star or blank is up to you:
> 
> Screen Shot 2022-10-10 at 9 21 24 AM
> 

Blank heading yeah, it should be coloured yellow too. Apparently I removed the face by accident.


> Follow-up question: should this star just be a svg like the other icons?

Questioned above in the thread, but for consistency should probably be an svg. Grab from font-awesome or something and making the fill colour be `#f79705`

**Comment by seanfarley** [comment]
Gotcha; doing that now

**Comment by seanfarley** [comment]
I extended the `defconst` logic to the other svg icons but noticed that the font size isn't what we want when the module loads. I suspect the star icon just happens to be the right size. For that reason, I think maybe it's not worth it to put star icon into its own const and instead follow what the other icons are doing.

File: exercism-modern.el [hunk 3]
```diff
@@ -73,6 +73,10 @@ Defaults to first entry in $PATH, can be overridden if required."
 (defvar exercism-modern-current-track nil
   "Current track to pull exercises for.")
 
+(defconst exercism-modern--star-svg
```
Comment (elken) [inline exercism-modern.el:76]:
```suggestion
(defcustom exercism-modern-star-icon (exercism-modern--load-from-package-root "icons/star.svg")
"Path to icon to use for recommended exercises."
:type 'file
:group 'exercism-modern)
```

File: exercism-modern.el [hunk 4]
```diff
@@ -73,6 +73,36 @@ Defaults to first entry in $PATH, can be overridden if required."
...
+  "Path to icon to use for difficulty level medium exercises."
+  :type 'file
+  :group 'exercism-modern)
+
+(defcustom exercism-modern-hard-icon (exercism-modern--load-from-package-root "icons/hard.svg")
+  "Path to icon to use for difficulty leve hard exercises."
+  :type 'file
+  :group 'exercism-modern)
+
+(defcustom exercism-modern-joined-icon (exercism-modern--load-from-package-root "icons/true.svg")
```
Comment (elken) [inline exercism-modern.el:96]: Currently just used for joined/unjoined but I can see it being used elsewhere so I'd rather they just be generic names like success/failure

File: exercism-modern.el [hunk 5]
```diff
@@ -232,11 +263,20 @@ Pass prefix BUFFER-PREFIX-ARG to prompt for a buffer instead."
                                            (is-unlocked (not (eq :json-false (alist-get 'is_unlocked exercise))))
                                            (is-recommended (not (eq :json-false (alist-get 'is_recommended exercise))))
                                            (text-face (if is-unlocked 'default 'shadow))
-                                           (difficulty-svg (exercism-modern--load-from-package-root (concat "icons/" difficulty ".svg"))))
+                                           (difficulty-svg (symbol-value (intern (concat "exercism-modern-" difficulty "-icon")))))
                                       (list slug
-                                            (vector (concat
-                                                     (when is-recommended
-                                                       (propertize "  " 'face 'exercism-modern-warning))
+                                            (vector (if is-recommended
```
Comment (elken) [inline exercism-modern.el:268]: when is-recommended (didn't I already commit that?)

File: exercism-modern.el (hunk 4 above)
Comment (seanfarley) [reply exercism-modern.el:96]: Fair enough

File: exercism-modern.el (hunk 5 above)
Comment (seanfarley) [reply exercism-modern.el:268]: But there's an `else` part for an empty string?

File: exercism-modern.el [hunk 6]
```diff
@@ -73,6 +73,45 @@ Defaults to first entry in $PATH, can be overridden if required."
...
+(defvar exercism-modern--load-path (file-name-directory load-file-name)
+  "Load path for the package so as to find media files.")
+
+(defun exercism-modern--load-from-package-root (path)
+  "Load PATH relative to the package directory."
+  (expand-file-name
+   path
+   (file-name-directory exercism-modern--load-path)))
+
+(defcustom exercism-modern-star-icon (exercism-modern--load-from-package-root "icons/star.svg")
```
Comment (elken) [inline exercism-modern.el:85]: Please group everything accordingly

customs -> vars -> fns

**Comment by elken** [comment]
Thanks for the contribution! :D

**Pull request merged**